src/genome.py

```diff
@@ -126,7 +143,20 @@ def copy_te(self, te: int, offset: int) -> int | None:
 
         If te is not active, return None (and do not copy it).
         """
-        ...  # FIXME
+        if te not in self.active:
+            return None
+        for i in range(len(self)):
+            if self.genome[i] == te:
+                start_orginal_te = i
+                break
+        start_copy = start_orginal_te + offset
+        while start_copy not in range(0, len(self)):
+            if start_copy > len(self):
+                start_copy = start_copy-len(self)
+            if start_copy < 0: 
+                start_copy = len(self) + start_copy
```

Comment:
If you have an overrun, `start_copy % len(genome)` will bring you back home to the right range. No need for these shenanigans.